### ISSUE TITLE
Build failure: aws-sam-cli 1.119.0

### ISSUE THREAD

**Issue by elct9620** [NixOS/nixpkgs#325990]
### Steps To Reproduce

1. nix profile install github:elct9620/nix-aura

> commit sha `6b9d39d0510402368965612a6d5a43835a2f9756`
> https://github.com/elct9620/nix-aura

### Build log

```
error: builder for '/nix/store/2yc5jy7d9ryfnh2l117hlvksnqw6si2q-aws-sam-cli-1.119.0.drv' failed with exit code 1;
       last 25 log lines:
       >     def _load_parts(self):
       > >       paths = self._raw_paths
       > E       AttributeError: 'PosixPath' object has no attribute '_raw_paths'
       >
       > /nix/store/3swy1vadi125g0c1vxqp8ykdr749803j-python3-3.12.4/lib/python3.12/pathlib.py:408: AttributeError
       > =============================== warnings summary ===============================
       > tests/unit/commands/local/cli_common/test_invoke_context.py::TestInvokeContext_local_lambda_runner::test_must_create_runner_using_warm_containers
       > tests/unit/lib/build_module/test_build_strategy.py::ParallelBuildStrategyTest::test_given_delegate_strategy_it_should_call_delegated_build_methods
       > tests/unit/lib/build_module/test_build_strategy.py::TestIncrementalBuildStrategy::test_assert_incremental_build_function_2
       > tests/unit/lib/utils/test_boto_utils.py::TestBotoUtils::test_get_boto_client_provider_with_config
       >   /nix/store/3swy1vadi125g0c1vxqp8ykdr749803j-python3-3.12.4/lib/python3.12/asyncio/base_events.py:726: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
       >     _warn(f"unclosed event loop {self!r}", ResourceWarning, source=self)
       >   Enable tracemalloc to get traceback where the object was allocated.
       >   See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED tests/unit/lib/build_module/test_app_builder.py::TestApplicationBuilder_update_template_windows::test_must_write_absolute_path_for_different_drives - AttributeError: type object 'WindowsPath' has no attribute '_from_parts'. D...
       > FAILED tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py::TestResourceLinking::test_resolve_resource_attribute_module_and_variable_references - AttributeError: 'has_calls' is not a valid assertion. Use a spec for the mo...
       > FAILED tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py::TestResourceLinking::test_resolve_resource_attribute_module_reference - AttributeError: 'has_calls' is not a valid assertion. Use a spec for the mo...
       > FAILED tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py::TestResourceLinking::test_resolve_resource_attribute_variable_reference - AttributeError: 'has_calls' is not a valid assertion. Use a spec for the mo...
       > FAILED tests/unit/lib/observability/cw_logs/test_cw_log_formatters.py::TestCWPrettyPrintFormatter::test_must_serialize_event - AttributeError: 'has_calls' is not a valid assertion. Use a spec for the mo...
       > FAILED tests/unit/lib/build_module/test_bundler.py::PostProcessHandler::test_check_invalid_lambda_handler - AttributeError: 'PosixPath' object has no attribute '_raw_paths'
       > =========== 6 failed, 6061 passed, 21 skipped, 4 warnings in 33.93s ============
       > /nix/store/89ghmy7qxyggz4sycc9qr1klinmrlgbx-stdenv-darwin/setup: line 1650: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix-store -l /nix/store/2yc5jy7d9ryfnh2l117hlvksnqw6si2q-aws-sam-cli-1.119.0.drv'.
error: 1 dependencies of derivation '/nix/store/zi93q1cxm6c2k65r8vishjigpib0xjzg-aura-cloud.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7hhsrpbp4x4jjp9i6b0czyj7pk4xc5nr-aura-full.drv' failed to build
```

### Notify maintainers

@lo1tuma @anthonyroussel

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.11 MiB unpacked):
  /nix/store/92slv99qfx8xqf2553aiv9xcb8g2imi6-DarwinTools-1
  /nix/store/1lzkqn0i09c181b6w9343jwlps6pswg1-nix-info
copying path '/nix/store/92slv99qfx8xqf2553aiv9xcb8g2imi6-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/1lzkqn0i09c181b6w9343jwlps6pswg1-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.21.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by elct9620** [comment]
It may fixed after #337137 and #303877 ready 🤔

**Comment by elct9620** [comment]
Fixed after update to latest
https://github.com/NixOS/nixpkgs/issues/337137#issuecomment-2333335284
